### PR TITLE
feat(dws): add datasource to query snapshots

### PR DESCRIPTION
**Pull request by wuzhuanhong**
**What this PR does / why we need it**:

Add new datasource to get the list of DWS snapshots within HuaweiCloud.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add new datasource.
2. add related document and acceptance test.
3. because it takes a long time to create a DWS cluster, an environment variable HW_DWS_CLUSTER_ID is provided as the DWS cluster ID to control the acceptance test.
```

## PR Checklist

* [x] Tests added/passed.

```
make testacc TEST='./huaweicloud/services/acceptance/dws' TESTARGS='-run TestAccDataSourceSn 
apshots'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dws -v -run TestAccDataSourceSn 
=== RUN   TestAccDataSourceSnapshots_basic
=== PAUSE TestAccDataSourceSnapshots_basic
=== CONT  TestAccDataSourceSnapshots_basic
--- PASS: TestAccDataSourceSnapshots_basic (310.08s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dws       310.155s
```

* [x] Documentation updated.
* [x] Schema updated.
* [ ] CheckDeleted.

  - **a. During query operation (Read Context)**
    aa. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<

    
  - **b. During delete/disassociate/unbind operation (Delete Context)**
    ba. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<

### CODE REVIEW

File: huaweicloud/services/acceptance/dws/data_source_huaweicloud_dws_snapshots_test.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  ]
+}
+
+locals {
+  cluster_ids = data.huaweicloud_dws_snapshots.test.snapshots[*].cluster_id
+}
+
+locals {
+  snapshot_ids = data.huaweicloud_dws_snapshots.test.snapshots[*].id
+}
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/dws/data_source_huaweicloud_dws_snapshots_test.go]:
```suggestion
locals {
  cluster_ids  = data.huaweicloud_dws_snapshots.test.snapshots[*].cluster_id
  snapshot_ids = data.huaweicloud_dws_snapshots.test.snapshots[*].id
}
```

**Comment by wuzhuanhong** [comment]
make testacc TEST='./huaweicloud/services/acceptance/dws' TESTARGS='-run TestAccDataSourceSnapshots'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dws -v -run TestAccDataSourceSnapshots -timeout 360m -parallel 4
=== RUN   TestAccDataSourceSnapshots_basic
=== PAUSE TestAccDataSourceSnapshots_basic
=== CONT  TestAccDataSourceSnapshots_basic
--- PASS: TestAccDataSourceSnapshots_basic (311.96s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dws       312.019s

File: huaweicloud/services/acceptance/dws/data_source_huaweicloud_dws_snapshots_test.go (hunk 1 above)
Comment (wuzhuanhong) [reply huaweicloud/services/acceptance/dws/data_source_huaweicloud_dws_snapshots_test.go]: fixed

File: docs/data-sources/dws_snapshots.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+* `region` - (Optional, String) Specifies the region in which to query the resource.
+  If omitted, the provider-level region will be used.
+
+## Attribute Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The data source ID.
+
+* `snapshots` - All snashots that match the filter parameters.
```
Comment (ruwenqiang123) [inline docs/data-sources/dws_snapshots.md:32]:
```suggestion
* `snapshots` - All snapshots that match the filter parameters.
```

File: docs/data-sources/dws_snapshots.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+* `id` - The ID of the snapshot.
+
+* `name` - The name of the snapshot.
+
+* `cluster_id` - The cluster ID corresponding of the snapshot.
+
+* `type` - The type of the snapshot.
+  + **MANUAL**
+  + **AUTOMATED**
```
Comment (ruwenqiang123) [inline docs/data-sources/dws_snapshots.md:47]:
```suggestion
  The value can be **MANUAL** or **AUTOMATED**.
```

File: docs/data-sources/dws_snapshots.md (hunk 2 above)
Comment (ruwenqiang123) [inline docs/data-sources/dws_snapshots.md]:
```suggestion
* `snapshots` - All snapshots that match the filter parameters.
```

File: docs/data-sources/dws_snapshots.md (hunk 3 above)
Comment (ruwenqiang123) [inline docs/data-sources/dws_snapshots.md:47]:
```suggestion
The value can be **MANUAL** or **AUTOMATED**.
```

Comment (Lance52259) [reply docs/data-sources/dws_snapshots.md:47]: unnecessary to adjust it, the current style just fine.

File: docs/data-sources/dws_snapshots.md (hunk 2 above)
Comment (wuzhuanhong) [reply docs/data-sources/dws_snapshots.md]: fixed

**Pull request merged**